lec01-introduction.Rmd

```diff
@@ -2,3 +2,171 @@
+
+```{r}
+ggplot(iris, aes(x=sepal_length, y=sepal_width, color=species)) +
+    geom_point()
+```
+
+Likewise, it would be really easy to make a histogram of the petal_length and color it according to the species.
+
+```{r}
+ggplot(iris, aes(x=petal_length, fill=species)) +
+    geom_histogram()
+```
+
+And we could get fancier by simply instructing R to split this plot in vertically stacked facets, or subplots, with one species in each facet.
+
+```{r}
+ggplot(iris, aes(x=petal_length, fill=species)) +
+    geom_histogram() +
+    facet_wrap(~ species, dir='v')
+```
+
+One last example If you don't have a clear idea of what you are looking for in a data set, maybe you are exploring publicly available data that you have not generated yourself, then it can be useful to visualize the relationships between all the variables to get an overview of the data.
+
+```{r}
+ggpairs(iris, aes(colour = species))
+```
+
+This plot is a bit overwhelming, and the purpose here is not to produce a publication ready figure. You can see that many of the labels are overlapping and not everything fits in the plotting window. We can tweak these properties, which we will get into later in the course. But even in its slightly rough form, this is a powerful data set overview for your own analyses.
+
+Hopefully, these examples have giving you a idea of how simple and powerful it is to use R for you exploratory data analyses. In a few lines of codes we have made some really nice plots, some of which are not even imaginable to create in Excel.
+
+## Saving data
+OK, now you should have a somewhat of an idea of how to generate program... But now that I am done with my analysis, how do I save it? First, what do we want to save? Definitely the plots, those are important. And remember that we said one of the biggest advantages over graphical software is that we have access to all the commands we typed in, and we can save them as an instruction manual for how to make these plots.
+
+Scripts are powerful for automation, but it is still a little bit messy in terms of saving you data, since we will keep the code in one text file and all the output graphs will be saved as separate images. It would not be clear exactly which part of the text file generated what images unless we added notes about this manually. Instead, it would be great if we could keep this all together, in sort of a notebook format... Luckily, there is something called the R Notebook (go to New -> R notebook), which allows us to do exactly that!
+
+# The R Notebook & R Markdown
```

Comment:
I would actually recommend introduction this before the plots, so you can type out and show them all together. And so you can create a html file at the end of the class.

Reply:
`s/#/###/`

Reply:
This section has a lot of text, which I think could be cut down. The less the better. I'd suggest pointing them to references/tutorials if they are interested, and also that we will be covering this in more detail as the course progresses. So there is no need to go into too much detail.

Reply:
Agreed; I don't think we need a "here's the long way to do it, but here's an even better way!" given the aim of this course.

Reply:
Agreed, I will make it ore focused on one way of doing things. Afraid I am running short on time soon though...